### PR TITLE
VPN-2452 - Implement subscription expiring message

### PR DESCRIPTION
**Pull request by brizental**
Had to reopen it for CI to be able to run fully.

Previous PR: https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5273

### CODE REVIEW

**Review by strseb — APPROVED** [review]
giphy

**Review by flodolo — APPROVED** [review]

File: tests/functional/testAddons.js [hunk 1]
```diff
@@ -5,126 +5,199 @@
...
 describe('Addons', function() {
   this.ctx.authenticationNeeded = true;
 
-  it('Empty addon index', async () => {
-    await vpn.resetAddons('01_empty_manifest');
-    await vpn.waitForCondition(async () => {
-      return parseInt(
-                 await vpn.getVPNProperty('VPNAddonManager', 'count'), 10) ===
-          0;
+  // it('Empty addon index', async () => {
```
Comment (bakulf) [inline tests/functional/testAddons.js:14]: Why?

File: tests/functional/testAddons.js [hunk 2]
```diff
@@ -5,126 +5,199 @@
...
+        mockDetails.subscription.current_period_end = expiresOn / 1000;
+        ctx.guardianSubscriptionDetailsCallback = () => {
+          ctx.guardianOverrideEndpoints.GETs['/api/v1/vpn/subscriptionDetails'].status = 200;
+          ctx.guardianOverrideEndpoints
+            .GETs['/api/v1/vpn/subscriptionDetails']
+            .body = mockDetails;
+        };
+
+        // Restart the VPN to load the new sub details.
+        await vpn.quit();
```
Comment (bakulf) [inline tests/functional/testAddons.js:147]: You don't need to do this. See the subscription test

File: tests/functional/setupVpn.js [hunk 3]
```diff
@@ -46,6 +46,8 @@ async function startAndConnect() {
   await vpn.connect(vpnWS, {hostname: '127.0.0.1'});
 }
 
+exports.startAndConnect = startAndConnect;
```
Comment (bakulf) [inline tests/functional/setupVpn.js:49]: You don't need this

**Review by bakulf — DISMISSED** [review]

File: tests/functional/testAddons.js (hunk 1 above)
Comment (brizental) [reply tests/functional/testAddons.js:14]: Oh I am doing testing, the test is failing on CI.

Comment (brizental) [reply tests/functional/testAddons.js:14]: Ignore my last commit.

**Comment by brizental** [comment]
Debugging notes for my tomorrow self: The subscription expiring addon is not being enabled or disabled, according to the logs. It is being loaded however.

File: src/apps/vpn/inspector/inspectorhandler.cpp [hunk 4]
```diff
@@ -567,6 +567,24 @@ static QList<InspectorCommand> s_commands{
                        return obj;
                      }},
 
+    InspectorCommand{"messages", "Returns a list of the loaded messages ids", 0,
```
Comment (bakulf) [inline src/apps/vpn/inspector/inspectorhandler.cpp:570]: We already have a similar method for guides. Wondering if we should return all the add-ons with types.

File: tests/functional/testAddons.js [hunk 5]
```diff
@@ -5,6 +5,8 @@
 const assert = require('assert');
 const queries = require('./queries.js');
 const vpn = require('./helper.js');
+const { SubscriptionDetails } = require('./servers/guardian_endpoints.js')
+const { startAndConnect } = require('./setupVpn.js')
```
Comment (bakulf) [inline tests/functional/testAddons.js:9]: This should not be needed.

**Review by bakulf — APPROVED** [review]

**Pull request merged**